### ISSUE TITLE
localization_spine produces empty segments on data the model is trained on

### ISSUE THREAD

**Issue by deepakri201** [Project-MONAI/MONAILabel#1267]
**Describe the bug**

Hi, 
I am trying to use the vertebrae_segmentation MONAI Label pipeline (inference) through 3DSlicer using the existing pretrained models. I've tried to run the first stage, localization_spine, but this produces segments that are empty. This occurs when I run inference on datasets from the training VERSE (both 2019 and 2020). 

**To Reproduce**

I've sucessfully set up and installed MONAI and MONAI Label locally, and can start the server and run the label app through Slicer. I've confirmed this all works by trying out the spleen segmentation pretrained model, which yields reasonable results. 
For the localization_spine, I've downloaded nifti files from VERSE 2019 and 2020 training data (which is the training data for the vertebrae_pipeline I believe). Therefore I expect the first step of the pipeline to perform well. 

1. monailabel start_server --app apps/radiology --studies apps/datasets/VERSE --conf models "localization_spine,localization_vertebra,segmentation_vertebra" 
2. Load the nifti data 
3. Select the localization_spine stage to run 

**Expected behavior**
I expect that the model that has been pretrained on VERSE to work well on the data it's been trained on. However, I could not find the exact patients that were used for training, nor if the dataset was from 2019 or 2020. 

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```
(monai) C:\Users\deepa>python -c "import monai; monai.config.print_debug_info()"
Printing MONAI config...
MONAI version: 1.1.0
Numpy version: 1.23.5
Pytorch version: 1.13.1
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: a2ec3752f54bfc3b40e7952234fbeb5452ed63e3
MONAI __file__: C:\Users\deepa\miniconda3\envs\monai\lib\site-packages\monai\__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.10
Nibabel version: 5.0.0
scikit-image version: 0.19.3
Pillow version: 9.4.0
Tensorboard version: 2.11.2
gdown version: 4.6.0
TorchVision version: 0.14.1
tqdm version: 4.64.1
lmdb version: 1.4.0
psutil version: 5.9.4
pandas version: 1.5.3
einops version: 0.6.0
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: 2.1.1
pynrrd version: 0.4.3

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

Printing system config...

System: Windows
Win32 version: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Win32 edition: Core
Platform: Windows-10-10.0.19045-SP0
Processor: Intel64 Family 6 Model 140 Stepping 1, GenuineIntel
Machine: AMD64
Python version: 3.10.8
Process name: python.exe
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: [popenfile(path='C:\\Windows\\System32\\en-US\\KernelBase.dll.mui', fd=-1), popenfile(path='C:\\Windows\\System32\\en-US\\kernel32.dll.mui', fd=-1), popenfile(path='C:\\Windows\\System32\\en-US\\tzres.dll.mui', fd=-1)]
Num physical CPUs: 4
Num logical CPUs: 8
Num usable CPUs: 8
CPU usage (%): [43.5, 25.6, 33.6, 23.1, 19.0, 14.0, 10.1, 14.3]
CPU freq. (MHz): 2803
Load avg. in last 1, 5, 15 mins (%): [0.0, 0.0, 0.0]
Disk usage (%): 63.7
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 15.8
Available memory (GB): 3.3
Used memory (GB): 12.5

Printing GPU config...

Num GPUs: 0
Has CUDA: False
cuDNN enabled: False

**Comment by tangy5** [comment]
Hi @deepakri201 , nice to see you here for follow ups. Can you paste MONAI Label logs here? 
SInce the segments are empty, that might because the model is not loaded correctly, we can find some trace in the monailabel logs.

**Comment by deepakri201** [comment]
> Hi @deepakri201 , nice to see you here for follow ups. Can you paste MONAI Label logs here? SInce the segments are empty, that might because the model is not loaded correctly, we can find some trace in the monailabel logs.

Hi Yucheng

Here is the MONAI label log: 

[monai_label_localization_spine_log.txt](https://github.com/Project-MONAI/MONAILabel/files/10537657/monai_label_localization_spine_log.txt)

Thank you!

**Comment by deepakri201** [comment]
> Hi @deepakri201 , nice to see you here for follow ups. Can you paste MONAI Label logs here? SInce the segments are empty, that might because the model is not loaded correctly, we can find some trace in the monailabel logs.

I tried localization_spine on a sample from medical decathalon (spleen dataset) and it seems to produce some segmentations, so I don't think it is an issue with the pretrained model: 

![example_localization_spine_on_spleen_dataset](https://user-images.githubusercontent.com/59979551/215543136-5b490a41-cf9b-4f1a-9db1-60d4ce348962.JPG)

**Comment by tangy5** [comment]
So is the data matters here? You can check whether image headers are different.

**Comment by deepakri201** [comment]
> 



> So is the data matters here? You can check whether image headers are different.

Yes it definitely seems to matter here, here are the headers: 

For the VERSE spine data the header is here: 
![verse_dataset_size](https://user-images.githubusercontent.com/59979551/215550939-0b4d1453-91f9-489d-9e97-f88f2513dd6c.JPG)

For the spleen data (decathalon) the header is here: 
![spleen_dataset_size](https://user-images.githubusercontent.com/59979551/215551071-db2d510f-489b-49c5-80ec-7953df706403.JPG)

I see that for the localization_spine the target_spacing is (1.3,1.3,1.3)

**Comment by tangy5** [comment]
For more references about the data, Andres mentioned here we can try the second stage localization_vertebra and see how that goes. 
Not all the VerSe dataset was used to train this model, some images from VerSe had wrong headers and create issues.

**Comment by tangy5** [comment]
Hi @deepakri201 , we had some very good discussion above, thanks, I will close for now, feel free to reopen or open another issue if there are more questions. Happy to help!
